app/models/library.rb

```diff
@@ -1,3 +1,18 @@
 class Library < ApplicationRecord
   has_many :books, dependent: :destroy
+
+  def book_count
+    self.books.count
+  end
+
+  def self.sort_by_book_count
+    libraries = Library.all
```

Comment:
Per our discussion, there is a way to do this with ActiveRecord but I would worry too much about it right now.